tests/test_downsample.py

```diff
@@ -153,20 +159,59 @@ def volume(emdb_vol, res, dtype):
     return vol
 
 
-def test_downsample_project(volume, res_ds):
+def test_downsample_project(volume, res_ds, legacy):
     """
     Test that vol.downsample.project == vol.project.downsample.
     """
     rot = np.eye(3, dtype=volume.dtype)  # project along z-axis
-    im_ds_proj = volume.downsample(res_ds).project(rot)
-    im_proj_ds = volume.project(rot).downsample(res_ds)
+    im_ds_proj = volume.downsample(res_ds, legacy=legacy).project(rot)
+    im_proj_ds = volume.project(rot).downsample(res_ds, legacy=legacy)
+
+    tol = 1e-09
+    if volume.dtype == np.float32:
+        tol = 1e-07
+    if legacy:
+        # project does not enforce legacy centering convention,
+        # so this property will not hold up to allclose tolerance.
+        tol = 1e-03
 
-    tol = 1e-07
-    if volume.dtype == np.float64:
-        tol = 1e-09
     np.testing.assert_allclose(im_ds_proj, im_proj_ds, atol=tol)
 
 
+def test_downsample_legacy(volume, res_ds):
+    """
+    The legay Matlab downsample method differs from ASPIRE-Python
+    downsample in that is uses a different centering convention,
+    off by a half pixel for odd images, and does not zero out
+    the nyquist frequency. By making these alterations to the
+    ASPIRE-Python downsampled images we can match legacy downsample
+    upt to `allclose`.
+    """
+    n_img = 10
+    dtype = volume.dtype
+    src = Simulation(
+        n=n_img,
+        vols=volume,
+        amplitudes=1,
+        dtype=dtype,
+        seed=1980,
+    )
+    ims = src.images[:]
+
+    # Legacy downsampled images.
+    ims_ds_legacy = ims.downsample(res_ds, legacy=True)
+
+    # ASPIRE-Python downsample with centering adjustments for odd resolution images.
+    shifts = 0.5 * np.ones((n_img, 2), dtype=dtype)
```

Comment:
Could we replace this with the formula that we found for the shifts? I'm worried that this is only half-pixel in certain special cases like this one.

Reply:
My notes from our meeting are a little incomplete, but what I have working locally for each case is this.
```
even -> even: no shift needed
even -> odd: shift by 1/2
odd -> odd: shift by 1/2 - res_ds/res_og
odd -> even: shift by -res_ds/res_og
```
I'm not getting `allclose` on the `odd -> even` case, so maybe I have something wrong here. For what it's worth, the original test works for different downsample ratios since the shifting occurs before and after downsampling depending on where the odd resolution is.

Reply:
[Edit Above: I had a typo in the formulas above... was missing a factor of 2 in the denominators.]

Yes, but the `1/2` pixel shift (on line 205) which happens prior to downsampling is converted in the downsampled image into a shift of `res_ds/(2 * res_og)` pixels by the act of downsampling. So the ratio ends up being accounted for.  I believe this is correct and have confirmed it for several different downsample ratios, but it's possible I'm misunderstanding something.

I suspect that I am not getting `allclose` for the last case due to some difference in our `Image.shift` method on even vs odd images.